source/geometry_model/initial_topography_model/ascii_data.cc

```diff
@@ -0,0 +1,182 @@
+          // No need to set the radial coordinate correctly,
+          // because it will be thrown away in get_data_component anyway
+          std_cxx11::array<double, dim> point;
+          point[0] = 6371000.0;
+          for (unsigned int d=0; d<dim-1; d++)
+            point[d+1] = surface_point[d];
+
+          global_point = Utilities::Coordinates::spherical_to_cartesian_coordinates<dim>(point);
+        }
+      else
+        AssertThrow(false, ExcNotImplemented());
+
+      const double topo = Utilities::AsciiDataBoundary<dim>::get_data_component(surface_boundary_id,
+                                                                                global_point,
+                                                                                0);
+
+      return topo;
+    }
+
+
+    template <int dim>
+    void
+    AsciiData<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Initial topography model");
+        {
+          Utilities::AsciiDataBase<dim>::declare_parameters(prm,
+                                                            "$ASPECT_SOURCE_DIR/data/geometry-model/initial-topography-model/ascii-data/test/",
+                                                            "box_2d_%s.0.txt");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    AsciiData<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Initial topography model");
+        {
+          Utilities::AsciiDataBase<dim>::parse_parameters(prm);
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace InitialTopographyModel
+  {
+    ASPECT_REGISTER_INITIAL_TOPOGRAPHY_MODEL(AsciiData,
+                                             "ascii data",
+                                             "Implementation of a model in which the surface "
+                                             "topography is derived from a file containing data "
+                                             "in ascii format. Note the required format of the "
+                                             "input data: The first lines may contain any number of comments "
+                                             "if they begin with '#', but one of these lines needs to "
+                                             "contain the number of grid points in each dimension as "
+                                             "for example '# POINTS: 3 3'. "
+                                             "The order of the data columns "
+                                             "has to be 'x', 'Topography [m]' in a 2d model and "
+                                             " 'x', 'y', 'Topography [m]' in a 3d model, which means that "
```

Comment:
are x, y really x/y coordinates, or are they interpreted as lat/long (for example) for spherical models?